usefulUtilities/hiFiCalendar/entityServerScripts/roomScheduleServer.js

```diff
@@ -0,0 +1,471 @@
+        var tzOffset = new Date().getTimezoneOffset() + tzHour * MINS_PER_HOUR + tzMin;
+        var date = new Date(Date.UTC(year, month - 1, day, hour, minute - tzOffset, second, msec));
+        return date;
+    };
+
+
+    // Update the calendar UI when we have the latest calendar information
+    this.postEvents = function(id, summary, start, end) {
+        if (id === that.entityID) {
+            var printedSchedule = '';
+
+            var startTimestamp = start;
+            var endTimestamp = end;
+
+            var tempEvent = {
+                summary: summary,
+                startTimestamp: startTimestamp,
+                endTimestamp: endTimestamp
+            };
+
+            that.room.eventList.push(tempEvent);
+
+            that.room.eventList.forEach(function(event) {
+                var startHours;
+                var endHours;
+
+                var eventStartHour = +event.startTimestamp.getHours();
+                var eventEndHour = +event.endTimestamp.getHours();
+                if (eventStartHour + that.timezoneOffset <= 0) {
+                    startHours = eventStartHour + that.timezoneOffset + NOON_HR;
+                } else if (eventStartHour + that.timezoneOffset <= (NOON_HR)) {
+                    startHours = eventStartHour + that.timezoneOffset;
+                } else {
+                    startHours = eventStartHour + that.timezoneOffset - NOON_HR;
+                }
+
+                if (eventEndHour + that.timezoneOffset <= 0) {
+                    endHours = eventEndHour + that.timezoneOffset + NOON_HR;
+                } else if (eventEndHour + that.timezoneOffset <= (NOON_HR)) {
+                    endHours = eventEndHour + that.timezoneOffset;
+                } else {
+                    endHours = eventEndHour + that.timezoneOffset - NOON_HR;
+                }
+
+                var startAmPm;
+                var endAmPm;
+
+                if (event.startTimestamp.getHours() + that.timezoneOffset < 0) {
+                    startAmPm = "pm";
+                } else if (event.startTimestamp.getHours() + that.timezoneOffset < (NOON_HR)) {
+                    startAmPm = "am";
+                } else {
+                    startAmPm = "pm";
+                }
+
+                if (event.endTimestamp.getHours() + that.timezoneOffset < 0) {
+                    endAmPm = "pm";
+                } else if (event.endTimestamp.getHours() + that.timezoneOffset < (NOON_HR)) {
+                    endAmPm = "am";            
+                } else {
+                    endAmPm = "pm";
+                }
+
+                printedSchedule = printedSchedule + 
+                event.summary +
+                "\n" +
+                startHours + 
+                ':' + 
+                (JSON.stringify(event.startTimestamp.getMinutes()).length > 1 ? 
+                    event.startTimestamp.getMinutes() :
+                    event.startTimestamp.getMinutes() + "0") + 
```

Comment:
good catch, this code here is actually a little weird.  It seems like it would get into a situation where if the length was 1 from 7:05 and you got 5, then you would get a 7:50.  Going to update date this to what i think it should be, taking your notes into consideration, and see if it works as expected.